model/riscv_insts_aext.sail

```diff
@@ -269,20 +225,9 @@ function clause execute (AMO(op, aq, rl, rs2, rs1, width, rd)) = {
                         AMOMINU => to_bits(sizeof(xlen), min(unsigned(rs2_val), unsigned(loaded))),
                         AMOMAXU => to_bits(sizeof(xlen), max(unsigned(rs2_val), unsigned(loaded)))
                       };
-                    let rval : xlenbits = match width {
-                      BYTE   => sign_extend(loaded[7..0]),
-                      HALF   => sign_extend(loaded[15..0]),
-                      WORD   => sign_extend(loaded[31..0]),
-                      DOUBLE => loaded
-                    };
-                    let wval : MemoryOpResult(bool) = match (width, sizeof(xlen)) {
-                      (BYTE, _)    => mem_write_value(addr, 1, result[7..0],  aq & rl, rl, true),
-                      (HALF, _)    => mem_write_value(addr, 2, result[15..0], aq & rl, rl, true),
-                      (WORD, _)    => mem_write_value(addr, 4, result[31..0], aq & rl, rl, true),
-                      (DOUBLE, 64) => mem_write_value(addr, 8, result,        aq & rl, rl, true),
-                      _            => internal_error(__FILE__, __LINE__, "Unexpected AMO width")
-                    };
-                    match (wval) {
+                    let rval : xlenbits = sign_extend(loaded[width_bytes * 8 - 1 .. 0]);
```

Comment:
Extending loaded on line 211 then re-extending it here is a bit confusing. Maybe just moving this to before line 211 would be enough to make it clearer? Or anything else you have in mind.

Reply:
Ah yeah I agree. This was just because I mechanically translated the code but after actually reading it I agree it was pretty confusing!

I've made an attempt at simplifying it by not extending the values and then later truncating them. I also changed the min/max ones to use `<_u` etc which I think is a bit nicer and plays better with the type system.